### PR TITLE
Normative: Reject an ambiguous time string even with a calendar

### PR DESCRIPTION
**Pull request by gibson042**
Fixes #2285

### CODE REVIEW

File: polyfill/lib/ecmascript.mjs [hunk 1]
```diff
@@ -375,7 +373,7 @@ export const ES = ObjectAssign({}, ES2020, {
     // and must be stripped so presence of a calendar doesn't result in interpretation
     // of otherwise ambiguous input as a time.
     const isoStringWithoutCalendar = calendar
-      ? $stringSlice(isoString, 0, isoString.length - calendar.length - 7)
+      ? ES.Call(StringPrototypeSlice, isoString, [0, -(calendar.length + 7)])
```
Comment (ljharb) [inline polyfill/lib/ecmascript.mjs:376]: why is this an improvement? this means the intrinsic cache isn't maximally primed by the dep graph.

Comment (gibson042) [reply polyfill/lib/ecmascript.mjs:376]: It's for consistency with the rest of the polyfill, which makes use of ES.Call for close alignment with the spec. This PR is not the right place to introduce callBind, although a PR dedicated to that might be appropriate.

Comment (ljharb) [reply polyfill/lib/ecmascript.mjs:376]: oh sorry, i missed that your PR was adding it in the first place, nvm.

Comment (gibson042) [reply polyfill/lib/ecmascript.mjs:376]: It also occurs to me that #2288 or an immediate followup is quite possibly the place.

**Review by ptomato — APPROVED** [review]
Agreed with the reasoning in #2285. Let's present this at the next TC39, marking as draft until then.

**Comment by ptomato** [comment]
This gained consensus at the July 2022 TC39 plenary. Tests are now available in https://github.com/tc39/test262/pull/3641. I had some last minute doubts about whether this one was actually correct, but I'm convinced by the discussion in #2285.

**Pull request merged**